### ISSUE TITLE
the way of loading teacher's logits may be wrong

### ISSUE THREAD

**Issue by HoJenson** [modelscope/easydistill#21]
<img width="1005" height="294" alt="Image" src="https://github.com/user-attachments/assets/74e2dd09-5de8-4184-8d2d-528830ff72da" />
<img width="1005" height="421" alt="Image" src="https://github.com/user-attachments/assets/375c1c0d-364c-4a4f-968f-728c79ce85a2" />

When you use global_step, after you have trained for several epochs, you may get an all-zero teacher logits because the start_idx and end_idx may be out of range. (you can print the length of loaded_data to verify this issue)

I think you can include the idx of the training data and use the idx to obtain the logits of the teachers.
